src/wallet/rpcwallet.cpp

```diff
@@ -2497,6 +2503,216 @@ UniValue fundrawtransaction(const UniValue& params, bool fHelp)
     return result;
 }
 
+UniValue getlabeladdress(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "getlabeladdress \"label\"\n"
+            "\nReturns the current 'label address' for this label.\n"
+            "\nArguments:\n"
+            "1. \"label\"       (string, required) The label for the address. It can also be set to the empty string \"\" to represent the default label.\n"
+            "\nResult:\n"
+            "\"bitcoinaddress\"   (string) The 'label address' for the label\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getlabeladdress", "")
+            + HelpExampleCli("getlabeladdress", "\"\"")
+            + HelpExampleCli("getlabeladdress", "\"mylabel\"")
+            + HelpExampleRpc("getlabeladdress", "\"mylabel\"")
+        );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    // Parse the label first so we don't generate a key if there's an error
+    string strLabel = AccountFromValue(params[0]);
+
+    UniValue ret(UniValue::VSTR);
+
+    ret = GetAccountAddress(strLabel).ToString();
+    return ret;
```

Comment:
It seems like L2529-2534 could be a one or two liner, rather than 4 (and maybe get rid of the whitespace).